src/client/clientlauncher.cpp

```diff
@@ -295,6 +292,16 @@ bool ClientLauncher::run(GameStartData &start_data, const Settings &cmd_args)
 		receiver->m_touchscreengui = NULL;
 #endif
 
+		/* Save the settings when leaving the game.
+		 * This makes sure that setting changes made in-game are persisted even in
+		 * case of an unclean shutdown (e.g. a crash).
+		 * This is useful on Android because closing the app from the
+		 * "Recents screen" results in an unclean shutdown.
+		 * Caveat: This means that the settings are saved twice when exiting Minetest.
+		 */
+		if (!g_settings_path.empty())
+			g_settings->updateConfigFile(g_settings_path.c_str());
```

Comment:
I added this one for symmetry. It's probably not as useful as the other one, but I can imagine a case where it's useful:

- Player does `/set time_speed = 0` or something in-game
- Player selects "Exit to Menu" in the pause menu, goes back to mainmenu (now the settings are saved)
- Player closes Minetest from the "Recents screen" => They would have lost their setting change if we hadn't saved the settings

That's quite unlikely to happen though, so I'm not sure if it's worth keeping.